fix(es): Resolve plugin paths using config path as base

titanous opened the pull request:
**Description:**

This makes relative paths work for plugin references, they will be relative to the location of the config that imports the plugin.

Previously relative paths did not work, as the resolver attempted to resolve using the plugin path as the base and the target, resulting in mangled paths.


**Related issue (if exists):** https://github.com/aspect-build/rules_swc/pull/149

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/swc-project/swc?pullRequest=6800) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/swc-project/swc?pullRequest=6800) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/swc-project/swc?pullRequest=6800) it.</sub>

kdy1 commented on crates/swc/src/config/mod.rs:
```diff
@@ -204,10 +204,6 @@ pub struct Options {
     #[serde(default = "default_swcrc")]
     pub swcrc: bool,
 
-    #[cfg(not(target_arch = "wasm32"))]
```
Please restore this

titanous replied:
Done.

titanous committed:
fix(es): Resolve plugin paths using config path as base

This makes relative paths work, they will be relative to the location of
the config that imports the plugin.

Previously relative paths did not work, as the resolver attempted to
resolve using the plugin path as the base and the target, resulting in
mangled paths.

kdy1 commented:
@kwonoj Thoughts?

kwonoj commented:
I am not sure if we want this. The reason I left the current experimental resolver to absolute only is, it is unclear which point we should use for the relative resolve. Having swcrc / config as implicit root may work, but also some cases may not work as expected. 

Having base to resolve in plugin context seems ok to do, but public-facing interface seems need some thought.

titanous commented:
In the case of `rules_swc`, Bazel does not provide a way for us to provide absolute paths (and there's no `node_modules` to climb to), so we need some sort of relative path mechanism for plugins. Alternatively it could be relative to the current working directory that swc is called from. Given that relative paths don't work at all currently, and this entire config section is marked experimental, would you be open to merging this and seeing how it works out?

titanous commented:
I'd also be fine with a CLI flag or config setting explicitly specifying where relative paths should be resolved from. Implicitly using the config location seemed like good UX so I went with it, but any of these options are a couple lines of code for me to get working in my build at this point so I don't have a preference :)

titanous commented:
Closing in favor of adding a `--plugin` flag to `swc_cli`, which will work even better for `rules_swc`.
